### PR TITLE
ci: Adding CI pipeline to verify windows provisioning script content BEFORE it gets check in

### PR DESCRIPTION
**Pull request by marosset**
…

**Reason for Change**:

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [x] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by marosset** [comment]
I have an azure devops job running on my fork for validation.
Once this merges I'll create a new devops job that triggers when anything in the staging folder is touched

File: .pipelines/pr-windows-signed-scripts.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+trigger: none
+pr: none
```
Comment (jsturtevant) [inline .pipelines/pr-windows-signed-scripts.yaml:2]: should this be on?

File: .pipelines/pr-windows-signed-scripts.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      -e WINDOWS_PROVISIONING_SCRIPTS_URL=${SCRIPT_PACKAGE_URL} \
+      -e USE_MANAGED_IDENTITY="false" \
+      ${DEIS_GO_DEV_IMAGE} make test-kubernetes
+    displayName: Run e2e tests
+
+  - script: |
+      SA_NAME="$(cat sa_name)" && \
+      az login --service-principal -u ${CLIENT_ID} -p "$(CLIENT_SECRET)" --tenant ${TENANT_ID} && \
+      az storage account delete -y -n ${SA_NAME} -g ${RESOURCE_GROUP_NAME}
+    displayName: Delete storage account
```
Comment (jsturtevant) [inline .pipelines/pr-windows-signed-scripts.yaml:60]: Should there be a `condition: always()` here to clean up the storage account if the e2e tests fail.

File: .pipelines/pr-windows-signed-scripts.yaml (hunk 1 above)
Comment (marosset) [reply .pipelines/pr-windows-signed-scripts.yaml:2]: No, we override these when we make the pipelines in Azure DevOps.

File: .pipelines/pr-windows-signed-scripts.yaml (hunk 2 above)
Comment (marosset) [reply .pipelines/pr-windows-signed-scripts.yaml:60]: yea, I can update that

**Comment by jsturtevant** [comment]
/lgtm

Addresses https://github.com/Azure/aks-engine/issues/4003

**Pull request merged**